hugo/content/docs/micro-layout.md

```diff
@@ -6,133 +6,157 @@ weight: 400
-    ],
-}
-
-// node0_ops
-<SCompartment> {
-    id: 'node0_ops',
-    type: 'comp:comp',
-    layout: 'vbox',
-    children: [
-        <SLabel> {
-            id: 'node0_op0',
-            type: 'label:text',
-            text: '+ foo(): integer'
-        }, 
-        <SLabel> {
-            id: 'node0_op1',
-            type: 'label:text',
-            text: '# bar(x: string): void'
-        }
-    ],
-}
+};
 ```
 
-Those three children are of type `SCompartment`, meaning that they can also implement a layout strategy for their children.
+This results in the following visuals:
 
-For example, the `node0_header` has a `hbox` layout, so its children will be arranged horizontally. Its child `node0_icon` has a `stack` layout and its children will be stacked on top of it.
-
-Now is a good time to modify the layout in the example and see how the rendering reacts!
-Alternatively, you could also implement your own micro-layout engine and inject it via [Dependency Injection](// link to dependency injection documentation)
+![layout-configuration](/assets/docs/layout-configuration.png)
 
+If you want different layout configurations, you can implement your own micro-layout engine and inject it via [Dependency Injection](// link to dependency injection documentation).
 
 ## Layout Options
 It is possible to fine-tune the micro-layout by using the property `layoutOptions`.
+```typescript
+<SNode> {
+    ...
+    layout: 'vbox',
+    layoutOptions: {
+        ...
+    }
+}
+```
 
 ### Alignment
 You can control how children are aligned with the following properties:
 * `hAlign: 'left' | 'center' | 'right'` for the horizontal alignment (not available for layout `hbox`)
+![hAlign](/assets/docs/layout-hAlign.png)
 * `vAlign: 'top' | 'center' | 'bottom'` for the vertical alignment (not available for layout `vbox`)
+![vAlign](/assets/docs/layout-vAlign.png)
 
 ### Padding
-Padding for a container is controlled by the following properties:
+Spacing between the container and its children is controlled by the following properties:
 * `paddingTop: number` for the padding at the top of the container (in pixels)
 * `paddingRight: number` for the padding at the right of the container (in pixels)
 * `paddingBottom: number` for the padding at the bottom of the container (in pixels)
 * `paddingLeft: number` for the padding at left top of the container (in pixels)
 * `paddingFactor: number` to define a factor for the padding depending on the size of the container
-
+![padding](/assets/docs/layout-padding.png)
 ### Size
-The size of a container is generally decided depending on the size of its children. However, we can control some aspects of it with:
+The micro-layout engine takes care of computing the size of a container depending on the size of its children. This can lead to unesthetic results, for example if the size of labels is very different between nodes. You can set a minimal size for a container with the following properties or even disable the resizing:
```

Comment:
just maybe this reads a little bit nicer
"This can lead to unesthetic results, if for example the size of labels differs by large amounts between nodes. To handle issues like these,  you can configure the size calculations through the following properties."